app/models/svie_user.rb

```diff
@@ -1,35 +1,49 @@
 class SvieUser
+
   def initialize(user)
     @user = user
   end
-  
-  def not_member?
-    @user.svie_state == 'NEMTAG'
+
+  def can_join?
+    !member? && !in_processing?
   end
 
   def member?
-    @user.svie_state == 'ELFOGADVA'
+    @user.svie_member_type != 'NEMTAG'
   end
 
   def in_processing?
-    @user.svie_state == 'FELDOLGOZASALATT'
+    !@user.svie_post_request.nil?
   end
 
   def inside_member?
-    @user.svie_member_type == 'RENDESTAG'
+    @user.svie_member_type == 'BELSOSTAG'
   end
 
   def outside_member?
-    @user.svie_member_type == 'PARTOLOTAG'
+    @user.svie_member_type == 'KULSOSTAG'
   end
 
   def inactive_member?
     @user.svie_member_type == 'OREGTAG'
   end
 
-  def remove_membership!
-    @user.svie_state = 'NEMTAG'
-    @user.svie_member_type = 'NEMTAG'
-    @user.save!
+  def can_join_to?(member_type)
```

Comment:
Ez a függvény lehet hogy megér egy kódkommentet hogy mi alapján működik (honnan hova lehet lépni), mert kicsit nehezen átlátható most